account-kit/react/src/components/auth/hooks/useOAuthVerify.ts

```diff
@@ -0,0 +1,56 @@
+"use client";
+import { useCallback } from "react";
+import { useAuthenticate } from "../../../hooks/useAuthenticate.js";
+import { useAuthContext, type AuthStep } from "../context.js";
+import type { AuthType } from "../types.js";
+
+export type UseOAuthVerifyReturnType = {
+  authenticate: (config: Extract<AuthType, { type: "social" }>) => void;
+  isPending: boolean;
+};
+
+export const useOAuthVerify = (): UseOAuthVerifyReturnType => {
+  const { authStep: _authStep, setAuthStep } = useAuthContext();
+  const authStep = _authStep as Extract<AuthStep, { type: "oauth_completing" }>;
+
+  const { authenticate: authenticate_, isPending } = useAuthenticate({
+    onMutate: () => {
+      setAuthStep({
+        ...authStep,
+      });
+    },
+    onError: (err) => {
+      console.error(
+        "authentication errror with authentication step",
+        err,
+        authStep
+      );
+      setAuthStep({
+        type: "oauth_completing",
+        config: authStep.config,
+        error: err,
+        ...(authStep.config.authProviderId === "auth0" && {
+          logoUrl: authStep.config.logoUrl,
+        }),
+      });
+    },
+    onSuccess: () => {
+      setAuthStep({ type: "complete" });
+    },
+  });
+
+  const authenticate = useCallback(
+    (config: Extract<AuthType, { type: "social" }>) => {
+      authenticate_({
+        ...config,
+        type: "oauth",
+      });
+    },
+    [authenticate_, authStep]
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useCallback has an unnecessary dependency: 'authStep'. Either exclude it or remove the dependency array.
<!-- __reviewdog__:ChA4NjJlZGMxYWVhNzViMDhkEgZyZGpzb24= -->